mirror_base.h

```diff
@@ -176,6 +161,23 @@ namespace mirror
 	template <typename T>
 	TypeDesc* GetTypeDesc(T) { return TypeDescGetter<T>::Get(); }
 
+	template <typename T>
+	struct TypeDescGetter<T, void, std::enable_if_t<std::is_enum<T>::value>>
+	{
+		static TypeDesc* Get()
+		{
+			switch (sizeof(T))
+			{
+			case 1: return TypeDescGetter<int8_t>::Get();
+			case 2: return TypeDescGetter<int16_t>::Get();
+			case 4: return TypeDescGetter<int32_t>::Get();
+			case 8: return TypeDescGetter<int64_t>::Get();
+			}
+			return nullptr;
+		}
+	};
+
+
```

Comment:
I dont really get why this had to move below GetTypeDesc, can you elaborate please ? is it still working as intended with that modification ?

Reply:
I just added a comment for the move reason just above. I don't know if it works, but the related errors disappear.